src/test/scala/com/microsoft/kusto/spark/keyVaultAuthenticationTest.scala

```diff
@@ -0,0 +1,62 @@
+package com.microsoft.kusto.spark
+
+import java.util.UUID
+
+import com.microsoft.azure.kusto.data.{ClientFactory, ConnectionStringBuilder}
+import com.microsoft.kusto.spark.datasource.KustoOptions
+import com.microsoft.kusto.spark.datasource.KustoOptions.SinkTableCreationMode
+import com.microsoft.kusto.spark.utils.KustoQueryUtils
+import org.apache.spark.sql.SparkSession
+
+import scala.collection.immutable
+import org.junit.runner.RunWith
+import org.scalatest.FlatSpec
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class keyVaultAuthenticationTest extends FlatSpec{
+  private val spark: SparkSession = SparkSession.builder()
+    .appName("KustoSink")
+    .master(f"local[1]")
```

Comment:
better to have at least[2] (I typically use [4]), otherwise it is just a single thread when you test locally